chapter04/rpi/ex3.pm

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+$all = {
+  Gilligan => { items => qw( all the stuff), location => 'The Island' },
+  Skipper => { items => qw( all the stuff), location => 'The Island' },
+  'Mr. Howell' => { items => qw( all the stuff), location => 'The Island' },
+  'Mrs. Howell' => { items => qw( all the stuff), location => 'The Island' },
+};
+
+sub check_items_for_all {
+  my ( $all ) = @_;
+
+  foreach $name ( keys %$all ) {
+    check_required_items( $name, $all->{name}->{items} );
+  
+    if ( $name =~ /Howell/ ) {
```

Comment:
I think the idea was just to update the islanders hash above, rather than to make this conditional.